src/components/core/dialog-confirmation/DialogConfirmation.tsx

```diff
@@ -0,0 +1,52 @@
+import Dialog from "@/components/core/dialog";
+import DialogConfirmationContent from "./DialogConfirmationContent";
+import type { DialogConfirmationProps } from "./types";
+
+export default function DialogConfirmation({
+  open,
+  children = <></>,
+  cancelText,
+  confirmText,
+  description,
+  title,
+  onClose,
+  onCancel,
+  onConfirm,
+  ...props
+}: DialogConfirmationProps) {
+  const onCancelHandler = (
+    e?: {},
+    reason?: "backdropClick" | "escapeKeyDown",
+  ) => {
+    onClose?.({}, "backdropClick");
+    onCancel?.(e, reason);
+  };
+
+  const onConfirmHandler = (
```

Comment:
All event handlers in this project should use `function` syntax rather than arrow functions.